src/datadog/trace_segment.cpp

```diff
@@ -337,5 +363,48 @@ void TraceSegment::inject(DictWriter& writer, const SpanData& span) {
   }
 }
 
+Expected<void> TraceSegment::extract(const DictReader& headers) {
+  if (!awaiting_delegated_sampling_decision_) return {};
+
+  if (auto sampling_delegation_response =
+          headers.lookup(DD_SAMPLING_DELEGATION_RESPONSE_HEADER)) {
+    auto sampling_decision =
+        parse_sampling_delegation_response(*sampling_delegation_response);
+    if (!sampling_decision) {
+      return sampling_decision.error();
+    }
+
+    // std::swap(sampling_decision_, *sampling_decision);
+    sampling_decision_ = *sampling_decision;
+    awaiting_delegated_sampling_decision_ = false;
+
+    if (spans_.front()->parent_id == 0) {
+      trace_tags_.emplace_back(tags::internal::sampling_decider, "0");
+    }
+  }
+
+  return {};
+}
+
+void TraceSegment::inject(DictWriter& writer) {
```

Comment:
We might need to keep track of whether sampling was delegated to us, and then only inject (or otherwise make it a no-op) if we know that we have been delegated to.

I need to think about this more...